packages/clients/textLocator/src/utils/textLocatorBackend/findLiterature/selectLiterature.ts

```diff
@@ -0,0 +1,95 @@
+import { PolarStore, SelectResultFunction } from '@polar/lib-custom-types'
+import { SearchResultSymbols } from '@polar/plugin-address-search'
+import { FeatureCollection } from 'geojson'
+import { searchCoastalGazetteerByToponym } from '../../coastalGazetteer/searchToponym'
+import { sorter } from '../../coastalGazetteer/responseInterpreter'
+import { GeometrySearchState } from '../../../plugins/GeometrySearch/types'
+import { LiteratureFeature, TitleLocationFrequency } from '../../../types'
+
+// NOTE hits (= feature.properties) and featureCollections are in sync; that is, hits[i] had findings featureCollections[i] in gazetteer
+const processLiteratureToponyms = (feature: LiteratureFeature) =>
+  function (
+    this: PolarStore<GeometrySearchState, GeometrySearchState>,
+    featureCollections: FeatureCollection[]
+  ) {
+    const { commit, dispatch } = this
+    const [featureCollection, titleLocationFrequencyChild] = Object.entries(
+      feature.properties
+    ).reduce(
+      (
+        [featureCollection, titleLocationFrequencyChild],
+        [toponym, count],
+        index
+      ) => {
+        if (featureCollections[index].features.length) {
+          const feature = featureCollections[index].features.sort(
+            sorter(toponym, 'title')
+          )[0]
+          titleLocationFrequencyChild[feature.id || ''] =
+            (titleLocationFrequencyChild[feature.id || ''] || 0) + count
+          featureCollection.features.push(feature)
+        }
+        return [featureCollection, titleLocationFrequencyChild]
+      },
+      [
+        { type: 'FeatureCollection', features: [] } as FeatureCollection,
+        {} as TitleLocationFrequency['string']['location_frequency'],
+      ]
+    )
+
+    commit('plugin/geometrySearch/setFeatureCollection', featureCollection, {
+      root: true,
+    })
```

Comment:
The frequencies are already present and are commited in the following lines, see variable `nextTitleLocationFrequency`; it's sync now instead of async here.

The data is now available in this setting from the start, as the backend API changed and includes them in this scenario.